### ISSUE TITLE
Расширение "Процедуры и функции"

### ISSUE THREAD

**Issue by DenVortEP** [BlizD/Tasks#324]
В расширении реализована подсистема которая позволяет: 
- Загружать из выгруженной в текст конфигурации экспортные методы общих модулей и объектов метаданных. Загруженные данные представлены в виде отдельных объектов "Процедуры и функции", подчиненные  модулям, которые в свою очередь подчинены конфигурации
 - Редактировать загруженные объекты для формирования структурированного комментария
 - Присваивать методам произвольные категории (области использования)
 - Осуществлять поиск методов по области использованию,  наименованиям методов, областей методов.
[ПроцедурыИФункцииОбщихМодулейКонфигурации_110.zip](https://github.com/BlizD/Tasks/files/14477259/_110.zip)

![1_ИнформацияОПодсистеме](https://github.com/BlizD/Tasks/assets/162089593/20d88eab-ded5-4524-83d2-30769584fb62)
![2_КИПодсистемы](https://github.com/BlizD/Tasks/assets/162089593/8b80ecf7-db87-41d5-a9cd-46bc8fe5fc26)
![3_НастройкаКаталогаКонфигурации](https://github.com/BlizD/Tasks/assets/162089593/072e0517-5457-4afa-a1cf-df354a5ce54f)
![4_Загрузка текстов модулей](https://github.com/BlizD/Tasks/assets/162089593/be7b846e-9bcf-4529-bebb-8baa0f668cef)
![5_СписокОбщихМодулей](https://github.com/BlizD/Tasks/assets/162089593/1fa59c90-9811-4123-9308-d009baa6c20c)
![6_СписокМодулейОбъектовМД](https://github.com/BlizD/Tasks/assets/162089593/9f85635e-4b31-405a-aeb6-154a38eafc1f)
![7_СписокМетодовОбщегоМодуля](https://github.com/BlizD/Tasks/assets/162089593/badc761e-b6aa-4068-85cf-3ed9894b0561)
![8_ОписаниеМетода](https://github.com/BlizD/Tasks/assets/162089593/d9045c95-03e5-4a40-a23f-b3751f88dc81)
![9_ОписаниеМетодаПараметры](https://github.com/BlizD/Tasks/assets/162089593/df9bca91-be2f-4d98-88e6-d451031e84bb)
![10_ОписаниеМетодаПараметры2](https://github.com/BlizD/Tasks/assets/162089593/5362106d-649c-434e-9285-feea022af6d9)
![11_СкомпелированныйКомментарий](https://github.com/BlizD/Tasks/assets/162089593/61e0c32f-9388-4d23-98ad-eb314f7cce1b)
![12_ПоискМетода](https://github.com/BlizD/Tasks/assets/162089593/3035e7eb-d2cd-4590-8b63-eb07bfda86b7)

**Comment by ViktorErmakov** [comment]
Было бы замечательно, хотя бы в репозитории продукта, описать примеры использования функционала, для чего он нужен, осветить кейсы. Я пока не понял для чего это нужно. Спасибо!

**Comment by DenVortEP** [comment]
Идея доработки возникла в результате сотого вопроса коллеги по работе: "А как называется та функция, которая позволяет..?".
В организациях с ограниченным ресурсом  времени на создание нормальной документации к разработанным решением часто не хватает, разработчики меняются быстро.  В результате возникают  конфигурации огромного объема. Чтобы найти тот или иной метод с нужным функционалом, требуется много времени (возможно я не знаю быстрых способов).  Как вариант, проще иногда разработать новый метод чем найти тот старый. 
Если у меня возникал такой вопрос, то приходилось лезть в глобальный поиск и искать  тематические фразы. Или вспоминать более общие методы, где использовались нужные. 
Доработка позволяет  уже найденную таким образом функцию отметить некоторой категорией, произвольной. Например "Строковая функция". При последующем поиске по фразе "Строк" она появится в списке найденных методов. Так же поиск можно осуществлять по нескольким кускам фраз в названиях методов или категориях. Например:  "Получить ТЗ" или "Таблиц выгруз".
Вторая возможность родилась уже при работе с методами и попыткой структурирования и описания того что есть.  При описании методов в комментариях, хотелось соответствовать рекомендациям 1С.  Редактировать вручную метод, который возвращает структуру с  большим количеством элементов, особенно если они  тоже структурные - сложно. Попытался упростить этот момент. В доработке есть возможность внести необходимые параметры, описания и получить  комментарий в соответствии с  рекомендациями 1С.
Сейчас после внедрения, обе возможности расширения активно используются. Так же как результат работы - были выявлены методы, дублирующие функционал.
На счет кейсов, правильно ли я понял, что мне в описании  этого расширения надо добавить  примеры использования  с последовательными шагами действий в этой же заявке под  кратким описанием?

**Comment by ViktorErmakov** [comment]
Да просто в репозитарии своем сделайте  описание в readme. Инструкция с картинками, можно видео записать с кейсами выложить там же ссылку.

**Comment by DenVortEP** [comment]
Виктор, добрый  день.
Я добавил расширение в репозиторий (https://github.com/DenVortEP/MethodSearch)
Попытался сделать то, о чем Вы просили.
Добавил в релизы последнюю версию расширения. Сообщите, пожалуйста, если что-то сделал не так. Это мой первый опыт публичной доработки и работы с GitHub.

**Comment by ViktorErmakov** [comment]
Да, вполне, спасибо)

**Comment by spspru** [comment]
Добрый день!
После загрузки расширения оно не отображается в интерфейсе клиента.
Роль АМ_ПользовательПроцедурПолные в Права доступа  Пользователю добавил - но все равно ничего не видно.

**Comment by DenVortEP** [comment]
Добрый день! 
Возможно у пользователя, которому вы добавили профиль с ролью из расширения установлены полные права. В этом случае Роль может не добавляться  пользователю.
Необходимо добавить роль в конфигураторе.

**Comment by spspru** [comment]
Да, спасибо, Вы правы - установил роль в Конфигураторе и заработало.

26 мая 2025 г., в 11:11, DenVortEP ***@***.***> написал(а):

﻿
[https://avatars.githubusercontent.com/u/162089593?s=20&v=4]DenVortEP left a comment (BlizD/Tasks#324)<https://github.com/BlizD/Tasks/issues/324#issuecomment-2908917077>

Добрый день!
Возможно у пользователя, которому вы добавили профиль с ролью из расширения установлены полные права. В этом случае Роль может не добавляться пользователю.
Необходимо добавить роль в конфигураторе.

**Comment by DenVortEP** [comment]
Рекомендую обновить до версии 1.1.2. Выложил сегодня.
Там исправлены некоторые ошибки.

**Comment by spspru** [comment]
Очень долго загружает модули.
Нет планов многопоточно через фоновые задания сделать?

И не смог найти кнопку открытия модуля целиком- полезно было бы.

Особенно если Монако для раскраски кода прикрутить. Есть планы такие?

26 мая 2025 г., в 14:13, DenVortEP ***@***.***> написал(а):

﻿
[https://avatars.githubusercontent.com/u/162089593?s=20&v=4]DenVortEP left a comment (BlizD/Tasks#324)<https://github.com/BlizD/Tasks/issues/324#issuecomment-2909358719>

Рекомендую обновить до версии 1.1.2. Выложил сегодня.
Там исправлены некоторые ошибки.

**Comment by DenVortEP** [comment]
У нас загрузка происходит не часто. Раз в неделю. 
Основная цель была - модули нетиповой конфигурации превратить  в объекты, для быстрого тематического поиска. Каждый раз запускать глобальный поиск, особенно если не помнишь, как та или иная функция называется не очень удобно. Какими -то иными средствами пользоваться не хотелось.  Этот задачник, очень емкий по функционалу и достаточен для небольшой команды разработки и при этом свободный в распространении. Поэтому и было принято решение на разработку расширения.
В условиях ограниченных трудовых ресурсов  совсем нет времени на правильное ведение документации. База старая, текста много. Чтобы не производить повторную работу по написанию общих методов хотелось сделать инструмент, в котором написал "таблиц" "знач" "загр"  и он тебе показал, что есть из экспортного по этой теме. После начала работы, добавили внутренний классификатор, для какой то группировки экспортных методов.  В истории разработки системы было много разработчиков. Никто себя не утруждал описанием или соблюдением каких -то правил разработки. Кроме этого, была задача не только компоновки и исключения дублирования, но и формирование описательной части. Механизм реализован, но на данный момент не используется из-за отсутствия времени. 

Каких то планов по расширению функционала пока нет. 
При работе, если обнаруживаются какие то  ошибки, их убираем.

**Comment by pruidzeko** [comment]
> У нас загрузка происходит не часто. Раз в неделю. Основная цель была - модули нетиповой конфигурации превратить в объекты, для быстрого тематического поиска.

А с расширениями - предусмотрена работа с ними в Вашей системе?

**Comment by DenVortEP** [comment]
Добрый день.
Расширение обрабатывает тексты выгруженной конфигурации в указанный каталог. Все обработанные данные привязывает к зарегистрированной базе в системе (Справочник "Конфигурации").
Я думаю, если Вы расширение зарегистрируете как отдельную конфигурацию, то после запуска обработки загрузки модулей и указании в качестве конфигурации вашего расширения, то все  загрузится и привяжется.
Такой варант не тестировался.  Работу производим в старых конфигурациях, не поддерживающих расширение.
